vpd-manager/src/parser.cpp

```diff
@@ -56,111 +58,144 @@ types::VPDMapVariant Parser::parse()
-            logging::logMessage(
-                "Input parameter type isn't compatible to update keyword's value on DBus for object path: " +
-                l_inventoryObjPath);
-            return -1;
-        }
 
-        // Get D-bus name for the given keyword
-        l_propertyName =
-            vpdSpecificUtility::getDbusPropNameForGivenKw(l_propertyName);
+            // Get D-bus name for the given keyword
+            l_propertyName =
+                vpdSpecificUtility::getDbusPropNameForGivenKw(l_propertyName);
 
-        // Create D-bus object map
-        types::ObjectMap l_dbusObjMap = {std::make_pair(
-            l_inventoryObjPath,
-            types::InterfaceMap{std::make_pair(
-                l_interfaceName, types::PropertyMap{std::make_pair(
-                                     l_propertyName, l_keywordValue)})})};
+            // Create D-bus object map
+            types::ObjectMap l_dbusObjMap = {std::make_pair(
+                l_inventoryObjPath,
+                types::InterfaceMap{std::make_pair(
+                    l_interfaceName, types::PropertyMap{std::make_pair(
+                                         l_propertyName, l_keywordValue)})})};
 
-        // Call PIM's Notify method to perform update
-        if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+            // Call PIM's Notify method to perform update
+            if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+            {
+                // Call to PIM's Notify method failed.
+                std::string l_errMsg("Notify PIM is failed for object path: " +
+                                     l_inventoryObjPath);
+                throw std::runtime_error(l_errMsg);
+            }
+        }
+
+        // Update keyword's value on redundant hardware if present
+        if (!l_redundantFruPath.empty())
         {
-            // Call to PIM's Notify method failed.
-            logging::logMessage("Notify PIM is failed for object path: " +
-                                l_inventoryObjPath);
-            return -1;
+            if (updateVpdKeywordOnRedundantPath(l_redundantFruPath,
+                                                i_paramsToWriteData) < 0)
+            {
+                std::string l_errMsg(
+                    "Error while updating keyword's value on redundant path " +
+                    l_redundantFruPath);
+                throw std::runtime_error(l_errMsg);
+            }
         }
-    }
 
-    // Update keyword's value on redundant hardware if present
-    if (!l_redundantFruPath.empty())
+        // TODO: Check if revert is required when any of the writes fails.
+        // TODO: Handle error logging
+    }
+    catch (const std::exception& l_ex)
     {
-        if (updateVpdKeywordOnRedundantPath(l_redundantFruPath,
-                                            i_paramsToWriteData) < 0)
+        std::string l_keywordIdentifier{};
+        if (const types::IpzData* l_ipzData =
```

Comment:
as all the throw has sufficient information,
do we need extra log here, printing l_ex.what() is enough ?

Reply:
The throws do not log the Record Name and Keyword Name.

Reply:
I am ok for the time being. Later this need to be moved to PEL.